lib/jwt.rb

```diff
@@ -163,17 +161,17 @@ def decode(jwt, key=nil, verify=true, options={}, &keyfinder)
     if options[:verify_not_before] && payload.include?('nbf')
       raise JWT::ImmatureSignature.new('Signature nbf has not been reached') unless payload['nbf'].to_i < (Time.now.to_i + options[:leeway])
     end
-    if options[:verify_iss] && payload.include?('iss')
-      raise JWT::InvalidIssuerError.new("Invalid issuer. Expected #{options['iss']}, received #{payload['iss']}") unless payload['iss'].to_s == options['iss'].to_s
+    if options[:verify_iss] && options['iss']
```

Comment:
this looks like it could lead to problems with implementers:

if the caller passes :verify_iss but not 'iss' (or the other way around), then this could leas to an unexpected successful verification.

Probably worth adding a check and failing with an argument error in the case that one is specified but not both.